corehq/apps/data_interfaces/interfaces.py

```diff
@@ -85,12 +87,22 @@ def rows(self):
     @property
     def report_context(self):
         context = super(CaseReassignmentInterface, self).report_context
-        active_users = self.get_all_users_by_domain(user_filter=tuple(HQUserType.all()), simplified=True)
+        if not self.request.can_access_all_locations:
+            accessible_location_ids = (SQLLocation.active_objects.accessible_location_ids(
+                self.request.domain,
+                self.request.couch_user)
+            )
+            accessible_user_ids = user_ids_at_locations(accessible_location_ids)
+            active_users = self.get_all_users_by_domain(user_filter=tuple(HQUserType.all()), simplified=True,
+                                                        user_ids=tuple(accessible_user_ids))
```

Comment:
> Moreover, is there a way to improve get_all_users_by_domain as well?
Haha, probably!  There may be some easy optimizations, I'm not sure.  That function also does way too much, in my opinion, I'd be interested to see if we could replace most usages with much simpler utilities (like `CommCareUser.by_domain`).